Beistellung_Implementierung/system/UML/ClassDiagram.puml

```diff
@@ -0,0 +1,353 @@
+        + ErrorHandler_setPrintCallback(PrintCallback) : void
+    }
+
+
+    Scheduler -[hidden]-> ErrorHandler
+    ErrorHandler -[hidden]-> SoftTimer
+
+}
+package "Service" #DDDDDD {
+    class Button {
+        + Button_init(void) : void
+        + Button_getState(ButtonID) : ButtonState
+    }
+
+    class Buzzer {
+        + Buzzer_init(void) : void
+        + Buzzer_beep(BuzzerID) : void
+    }
+
+    class Display {
+        + Display_init(void) : void
+        + Display_clear(void) : void
+        + Display_clearLine(void) : void
+        + Display_home(void) : void
+        + Display_gotoxy(UInt8, UInt8) : void
+        + Display_write(char*, UInt8) : void
+        + Display_writeBar(UInt8) : void
+    }
+
+    class LED {
+        + Led_init(void) : void
+        + Led_switchOn(Led_Led) : void
+        + Led_switchOff(Led_Led) : void
+    }
+
+    class DriveControl {
+        + DriveControl_init(void) : void
+        + DriveControl_drive(DriveControlMotorID, UInt8, DriveControl_Direction) : void
+        + DriveControl_getMileage(void) : Int32
+        + DriveControl_resetMileage(void) : void
+    }
+
+    class LineSensor {
+        + LineSensor_init(void) : void
+        + LineSensor_startCalibration(void) : void
+        + LineSensor_stopCalibration(void) : void
+        + LineSensor_getCalibrationState(void) : bool
+        + LineSensor_read(LineSensorValues*) : void
+        + LineSensor_enableEmitter(void)
+        + LineSensor_disableEmitter(void) : void
+    }
+}
+
+package "HAL" #DDDDDD {
+
+    class GPIO{
+        + Gpio_init(void) : Gpio_Ret
+        + Gpio_write(Gpio_ID, Gpio_State) : Gpio_Ret
+        + Gpio_read(Gpio_ID, Gpio_State* : Gpio_Ret
+        + Gpio_alloc(Gpio_ID, UInt8*) : Gpio_Ret
+        + Gpio_free(Gpio_ID, UInt8) : Gpio_Ret
+    }
+    class PWM{
+        + Pwm_init(void) : void
+        + Pwm_setDutyCycle(PwmID, UInt8) : void
+    }
+    class TickTimer{
+        + TickTimer_init(void) : void
+        + TickTimer_setCallback(TimerTickCallback) : void
+        + TickTimer_get(void) : UInt64
+        + TickTimer_delay(UInt8) : void
+    }
+    class IRQ{
+        + Irq_init(void) : void
+        + Irq_enable(IrqID) : void
+        + Irq_disable(IrqID) : void
+        + Irq_setCallback(IrqID, IrqCallback) : void
+    }
+}
+    class main {
+
+    }
+    main ..> Scheduler
+
+}
+
+package "Application"  {
+
+    class "MainTask" {
+        -gMainTask: Task
+        -mainTaskWork (void * data): void
+        + MainTask_init(void) : MainTask_Ret
+    }
+
+    Scheduler <.up.. MainTask : register
+
+    enum EventEnum {
+        NO_EVENT_HAS_HAPPEND = 0,
+        INIZALIZATION_DONE,
+        CALIBRATION_DONE,
+        COUNTDOWN_IS_FINISHED,
+        POWER_TO_THE_MOTORS_HAS_BEEN_STOPPED,
+        START_BUTTON_HAS_BEEN_RELEASED,
+        CALIBRATION_BUTTON_HAS_BEEN_RELEASED,
+        PARAM_BUTTON_HAS_BEEN_RELEASED,
+        START_FINISH_LINE_WAS_RECOGINZED,
+        GUIDELINE_WAS_RECOGNIZED,
```

Comment:
named differently in the Code
```suggestion
        THE_GUIDELINE_WAS_RECOGNIZED,
```

Reply:
It's also called GUIDELINE_WAS_RECOGNIZED in the code now.